assets/js/modules/analytics-4/components/dashboard/ActivationBanner/SetupBanner.js

```diff
@@ -51,6 +55,16 @@ export default function SetupBanner( { onCTAClick } ) {
 		select( CORE_USER ).getReferenceDate()
 	);
 
+	const { submitChanges } = useDispatch( MODULES_ANALYTICS_4 );
+
+	const handleSubmitChanges = useCallback( async () => {
+		const { error } = await submitChanges();
+
+		if ( error ) {
+			setErrorNotice( error );
+		}
+	}, [ submitChanges ] );
```

Comment:
This is the code you added as an action, but `submitChanges` on its own won't do anything if there are no changes made, which is the case here when there is no GA4 properties setup and the user doesn't have a `propertyID`/`webDataStreamID` like you mentioned.

From the [Design Doc](https://docs.google.com/document/d/1DeWo38lcV7lvLFfcmt-mQ0iaxzB4qfiPArs_yZzYkIM/edit#) and [Figma Design](https://www.figma.com/file/vMaCWwr6lpk4PrJWb7jIpz/GA4-Banner-Input?node-id=1082%3A1549), it doesn't look like there's any specific UI for this, so the "Create Property" should do whatever the Settings page in Analytics does to create a new property. That means clicking it, UX-wise, should prompt the user for extra permissions and create a GA4 property by redirecting them to Analytics I think.

Whatever the Analytics Settings component already does can be used here I think.

Maybe @techanvil can confirm since he wrote up the Design Doc and has more context, but that should do the trick.

Reply:
Hi @tofumatt & @nfmohit - it looks like an implementation detail that is mentioned in the Feature Description has been missed along the way here. This does correlate with what the `PropertySelect` component is already doing:

> For the "no existing GA4 property" variants, i.e when the CTA is Create property, ensure the GA4 `propertyID` is set to `PROPERTY_CREATE` prior to calling `submitChanges`.

Dispatching `selectProperty( PROPERTY_CREATE )` will ensure the execution path to create the property will be taken within `submitChanges`.

https://github.com/google/site-kit-wp/blob/71681690c6faf941872d3a8392313e5b027d7907/assets/js/modules/analytics-4/datastore/settings.js#L57-L61

Make sure to only dispatch this action for the "no existing GA4 property" variants:
https://github.com/google/site-kit-wp/blob/71681690c6faf941872d3a8392313e5b027d7907/assets/js/modules/analytics-4/components/dashboard/ActivationBanner/SetupBanner.js#L117-L141

I'd refactor it slightly so it's:

```js
} else {
  selectProperty( PROPERTY_CREATE );

  if ( existingTag ) {
    // ...
  } else {
    // ...
  }
}
```

One more aspect to be aware of is that the case where the _edit_ scope is not available is handled in https://github.com/google/site-kit-wp/issues/5278. So this issue only really needs to handle the case where the permission already exists.

In terms of UX the expectation is for the property creation to happen in the background while the spinner is active, it's not necessary to redirect to Analytics for this, although of course a redirect to the OAuth flow will be necessary to grant permissions but as mentioned this will be handled via #5278.

Reply:
Thank you, @tofumatt and @techanvil! I have updated this PR to set the GA4 `propertyID` to `PROPERTY_CREATE` for no existing GA4 property. The spinner now shows up to create a measurement ID, but shows an OAuth flow or a permissions error, which I believe will be handled in #5278.